src/IssueViz.Security/Store/HotspotsStore.cs

```diff
@@ -72,6 +86,15 @@ private void NotifyHotspotChanged(IAnalysisIssueVisualization changedHotspot)
             IssuesChanged.Invoke(this, new IssuesChangedEventArgs(hotspotFilePaths));
         }
 
+        private IAnalysisIssueVisualization Exists(IAnalysisIssueVisualization hotspot)
+        {
+            var matches = Hotspots
+                .ToLookup(x => ((IHotspot)x.Issue).HotspotKey)
```

Comment:
Is the lookup necessary - wouldn't a simple `FirstOrDefault` on the collection do? e.g.

```
            var key = ((IHotspot)hotspot.Issue).HotspotKey;
            return Hotspots.FirstOrDefault(x => ((IHotspot)x.Issue).HotspotKey == key);
```

Reply:
You're right; I was originally implementing IEquatable, then changed the lookup, and completely missed the obvious solution. Thanks!

Reply:
Could be worse; when I thought about this yesterday, I missed the fact that there was a unique `HotspotKey` and thought we'd have to compare the issue fields...

Reply:
That's exactly what I did originally...

Reply:
🤦‍♂️